e2e/tests/node.go

```diff
@@ -263,3 +261,24 @@ func formatURLString(serviceAlias, rpcUrl, path string) string {
 	}
 	return fmt.Sprintf("http://%s.%s/v1/%s", serviceAlias, rpcUrl, path)
 }
+
+// getVirtualHostFromHost returns the given rpcUrl host stripped of the port
+// and with the service included as a subdomain to be used in the "Host" request header.
+// This is needed by the current DevNet infrastructure that does not support arbitrary
+// subdomains which the PATH Gateway needs to identify the requested service.
+func getVirtualHostFromHost(rpcUrl *url.URL, service string) string {
+	// Store the virtual host we want to send in the Host header
+	virtualHost := rpcUrl.Host
+
+	if len(service) > 0 {
+		// Strip port if it exists and add service prefix
+		host, _, err := net.SplitHostPort(rpcUrl.Host)
+		if err != nil {
+			// No port in the host, use as-is
+			host = rpcUrl.Host
+		}
+		virtualHost = fmt.Sprintf("%s.%s", service, host)
+	}
+
+	return virtualHost
+}
```

Comment:
```suggestion
// getVirtualHostFromUrlForService returns a virtual host taking into consideration
// the URL's host and the service if it's non-empty.
// Specifically, it:
// 	1. Extract's the host from the rpcURL
//      2. Prefixes the service as a subdomain to (1)the given rpcUrl host stripped of the port
// TODO_FUTURE: This is needed as of PR #879 because the DevNet infrastructure does
// not support arbitrary subdomains due to TLS certificate-related complexities.
func getVirtualHostFromUrlForService(rpcUrl *url.URL, service string) string {
	// Store the virtual host we want to send in the Host header
	virtualHost := rpcUrl.Host

	if len(service) > 0 {
		// Strip port if it exists and add service prefix
		host, _, err := net.SplitHostPort(rpcUrl.Host)
		if err != nil {
			// err is non-nil if rpcUrl.Host does not have a port.
			// Use the entire host as is
			host = rpcUrl.Host
		}
		virtualHost = fmt.Sprintf("%s.%s", service, host)
	}

	return virtualHost
}
```